system-addon/test/functional/mochitest/browser_highlights_section.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+
+async function add_highlights_bookmark(count) {
+  const bookmarks = new Array(count).fill(null).map((entry, i) => ({
+    parentGuid: PlacesUtils.bookmarks.unfiledGuid,
+    title: "foo",
+    url: `https://mozilla${i}.com/nowNew`
+  }));
+
+  for (let placeInfo of bookmarks) {
+    await PlacesUtils.bookmarks.insert(placeInfo);
+  }
+
+  // Add bookmark visits
+  for (let placeInfo of bookmarks) {
+    await PlacesTestUtils.addVisits(placeInfo.url);
+  }
+}
+
+// Test populating highlights section with bookmarks.
```

Comment:
I like that you added comments; I find a lot of mochi tests in the tree are really under-documented and hard to read/debug as a result. I wonder if it might be a good idea to write them in the same style as we do for our unit tests, i.e.

"it should populate the highlights section with each bookmark that is added"

Any thoughts @dmose @Mardak?

Reply:
@k88hudson I've kept the comments previously added at the top of the test which are more general and added more detailed ones where we actually make the assertions since some tests have more than 1 and the expectations are different.

Reply:
The "it should ..." style is actually really nice. 👍